common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -102,3 +109,171 @@ def perform_destroy(self, instance):
+        UUID so the course_run_id has to correspond to the Course that is assigned to
+        the Entitlement.
+
+        When this API is called for a user who is already enrolled in a run that User
+        will be unenrolled from their current run and enrolled in the new run if it is
+        available.
+        """
+        course_run_id = request.data.get('course_run_id', None)
+
+        if not course_run_id:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data='The Course Run ID was not provided.'
+            )
+
+        # Verify that the user has an Entitlement for the provided Course UUID.
+        try:
+            entitlement = CourseEntitlement.objects.get(uuid=uuid, user=request.user, expired_at=None)
+        except CourseEntitlement.DoesNotExist:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data='The Entitlement for this UUID does not exist or is Expired.'
+            )
+
+        # Verify the course run ID is of the same type as the Course entitlement.
+        course_run_valid = self._verify_course_run_for_entitlement(entitlement, course_run_id)
+        if not course_run_valid:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    'message': 'The Course Run ID is not a match for this Course Entitlement.'
+                }
+            )
+
+        # Determine if this is a Switch session or a simple enroll and handle both.
+        try:
+            course_run_string = CourseKey.from_string(course_run_id)
+        except InvalidKeyError:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    'message': 'Invalid {course_id}'.format(course_id=course_run_id)
+                }
+            )
+        if entitlement.enrollment_course_run is None:
+            response = self._enroll_entitlement(
+                entitlement=entitlement,
+                course_run_key=course_run_string,
+                user=request.user
+            )
+            if response:
+                return response
+        elif entitlement.enrollment_course_run.course_id != course_run_id:
+            self._unenroll_entitlement(
+                entitlement=entitlement,
+                course_run_key=entitlement.enrollment_course_run.course_id,
+                user=request.user
+            )
+            response = self._enroll_entitlement(
+                entitlement=entitlement,
+                course_run_key=course_run_string,
+                user=request.user
+            )
+            if response:
+                return response
+
+        data = self.get_serializer(entitlement).data
+        data['course_run_id'] = course_run_id
+
+        return Response(
+            status=status.HTTP_201_CREATED,
+            data=data
```

Comment:
This is unexpected to me.  This endpoint is dealing with enrollments not entitlements.  I don't think that we should be returning an entitlement here.

I think the way you had it before was fin, returning en enrollment object with `entitlement_uuid`, `course_run_id`, and `active` flag. 

The true right way is to create an `EntitlementEnrollmentSerializer` and wire it up like this.  For expediency I think returning the hand crafted dict is fine too.  Given that all this exists it may be easier to "fix forward" and just create the serializer anyway

Reply:
I can return it to that.  We can discuss this in more detail tomorrow.

Reply:
Also, I do not need to send the is_active flag anymore. That was a remnant of our initial implementation.